tests/unittests/hdf5_import_tests.py

```
@@ -0,0 +1,451 @@
+        dataset.delete()
+
+    def test_from_dir_auto_images_dir(self):
+        """Test that images_dir is auto-created when not specified."""
+        _create_hdf5_pixel_images(self._hdf5_path, num_images=3)
+
+        dataset = fo.Dataset.from_dir(
+            dataset_type=fo.types.HDF5Dataset,
+            hdf5_path=self._hdf5_path,
+        )
+
+        self.assertEqual(len(dataset), 3)
```

Ruff (0.15.2)

[warning] Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)